plugins/pulp_python/plugins/models.py

```diff
@@ -1,120 +1,150 @@
-from gettext import gettext as _
 import hashlib
-import re
-import tarfile
+import os
 
 from mongoengine import StringField
+import pkg_resources
 from pulp.server.db.model import FileContentUnit
+from twine.package import PackageFile  # noqa
 
 from pulp_python.common import constants
+from pulp_python.plugins import querysets
 
 
 DEFAULT_CHECKSUM_TYPE = 'sha512'
-
-
-# These are required to be in the PKG-INFO file.
-REQUIRED_ATTRS = ('name', 'version', 'summary', 'home_page', 'author', 'author_email', 'license',
-                  'description', 'platform')
+PACKAGE_ATTRS = ('author', 'name', 'packagetype', 'summary', 'version')
 
 
 class Package(FileContentUnit):
     """
     This class represents a Python package.
-    """
 
-    # unit key
-    name = StringField(required=True)
-    version = StringField(required=True)
+    Packages stored on PyPI have significantly more metadata associated with them, but we have
+    chosen to keep this model minimal. Below is an explanation for why this was done.
+
+    There are some overloaded terms that should be briefly discussed. We will be using the
+    semantics of PEP 426. https://www.python.org/dev/peps/pep-0426/#id14
+
+    Package - An individual, installable Python package. Uniqueness of a package is determined by
+              the project name, target architecture, and Python version. Note that there may not be
+              a target architecture in the case of architecture agnostic packages. A Python version
+              is also optional if there are no requirements for a specific runtime version. There
+              are two accepted package types, sdist and wheel.
+    Release - A snapshot of the project. A particular release of the project will all share the
+              same version number, but there may be multiple packages for a given release for
+              different architectures, Python versions, and package types.
+    Project - The entire group of packages of all releases.
+
+    The way that PyPI stores the metadata on their end leaves this model open to potential
+    inaccuracies because they assume that all packages of the same project share certain metadata,
+    including mutable fields. In the situation where a mutable field changes between releases, only
+    the latest version is present in the metadata accessible over the API. This is could be
+    particularly problematic for a field like "license". For this reason, most of these fields have
+    been left out of this model. The only fields that are included here are necessary for sync
+    (name, version, filename, path) or are useful for disambiguation between similarly named
+    projects (author, summary). By keeping the scope of the metadata as small as is reasonable, we
+    have less complexity and reduced risk of innacuracy.
```

Comment:
I made a note on ``package_specific_metadata`` about documenting the section of metadata it provides. It might be worth documenting what the expected PyPI JSON is for this model.